### DISCUSSION TITLE
Should we move quill_delta_from_html to the repo?

### DISCUSSION THREAD

**Discussion by CatHood0** [singerdmx/flutter-quill#2006] [General]
As mentioned by @EchoEllet in https://github.com/singerdmx/flutter-quill/pull/1984#issuecomment-2212598291 it could be a good option to directly move `flutter_quill_delta_from_html` for the following reasons

**Greater ease of accessing the code**: Moving the package to the repository makes access easier for all developers, not only the Flutter Quill team but also for curious developers who want to understand or improve some of the already existing features of `quill_delta_from_html` and allows greater simplicity to make a PR.

**Testing automation**: `flutter_quill_delta_from_html` contains a fairly wide and diverse variety where many use cases that may come from users are tested. This makes it cumbersome to have to manually test all use cases. By moving it, we make testing much easier thanks to its automation.

**Greater support from the Flutter Quill team**: Since the package was created by an external person, it makes it quite complicated not only to access the code, but also to handle errors reported in the package, since not being directly inside `flutter_quill` has to be moved to an external repo and notified about the error, which is a bit slow. Moving it decreases the chances that the package will no longer be maintained by the creator since it is not directly dependent on a single person.

So, even with the aforementioned, should we keep it as an external dependency or move it to the package?

**Comment by EchoEllet** [comment]
Thank you for the suggestion.

General, I'm with any new features that help the developers and don't affect the existing projects that might not use the new features. 

The only minor issue is that issues regarding the HTML converter will be created on the same repository without having a way to filter the issues, I'm no longer a contributor to the project, I only do minor and quick tasks, though the other contributors might be only interested in the editor itself.

I see that you're fixing issues quickly as soon as they are created, though the issue might still be there if the contributor searches for all issues, including solved ones.

We have two ways to solve this while still making this package part of the repo:

1. Either move all issues related to HTML to its own repo by providing a link.
2. Or require a new input in the GitHub bug form template and use it to add a label so we can filter the issues.

Other than that, we should be good to go. We're not required to fulfill this suggestion.

> it makes it quite complicated not only to access the code, but also to handle errors reported in the package, since not being directly inside flutter_quill has to be moved to an external repo and notified about the error, which is a bit slow.

If the main goal of the move is to get the fixes for the newer version more easily, we can automate this task by making the CI update the package to the latest version which will also update the minimum required version, we're not required to link the two repositories with a GitHub token or having a repo read info about the other, we can simply use Dart CLI.

The name `flutter_quill_delta_from_html` might be already good since Flutter Quill could indicate the project name itself and not if it's Flutter specific, though it could also be confusing.

Are you interested in making the [quill_delta_from_html](https://pub.dev/packages/quill_delta_from_html) package part of the automated publishing on this repo? This will make the version of the HTML package the same as the other packages (1.3.1 -> 9.5.15) or only make it part of the repository without having the same package version (1.3.1)?

I still suggest avoiding exposing the HTML package directly from `flutter_quill` and using it internally.

You can use the approach that's more suitable.

In either case, thank you for your time and efforts.

**Comment by CatHood0** [comment]
> The only minor issue, is that issues regarding the `HTML` converter will be created on the same repository without having a way to filter the issues, I'm no longer a contributor of this project, I only do minor and quick tasks, though the other contributors might not be interested in `HTML` and they might be interested in the editor itself.

It is true, filtering the problems until now has been quite complicated since the labels have been not very intuitive in terms of describing the problem. It would be good to have more options.

> All issues related to `HTML` to its own repo by providing a link.

I can't deny that this seems better to me than directly deploying the package inside `flutter_quill`. Even with the benefits that this would bring, it seems better to me to keep it in an external repository since it is easier to solve these problems that are not always related to important problems that affect the user themselves when using the package.

**Reply by EchoEllet** [reply]
> It would be good to have more options.

The only minor issue it requires more input from the developer, or we could either automate it or manually add the label.

**Reply by EchoEllet** [reply]
Moving the package also makes it a bit harder for you or other contributors to focus on issues related to HTML or to the editor itself, though, if you're okay with this.

I don't see any reasons preventing us from the move.

**Comment by EchoEllet** [comment]
Noticed the [flutter_quill_delta_from_html](https://pub.dev/packages/flutter_quill_delta_from_html) depends on the `flutter` package

https://github.com/CatHood0/flutter_quill_delta_from_html/blob/0a18d767321ce5e5a477290253ea7e5553f7a6a4/lib/parser/html_utils.dart#L1

https://github.com/CatHood0/flutter_quill_delta_from_html/blob/0a18d767321ce5e5a477290253ea7e5553f7a6a4/pubspec.yaml#L13-L14

The description says it's a Dart package

> This is a Dart package that converts HTML input into Quill Delta format, which is used in the [Quill Js](https://quilljs.com/) package.

Fixing this issue should be trivial.

**Reply by CatHood0** [reply]
Thanks! I don't notice about it
